tofu/geom/_GG02.pyx

```diff
@@ -3330,52 +2965,461 @@ def LOS_calc_signal2(func, double[:,::1] Ds, double[:,::1] us, dL,
+                                                         &loc_eff_res[0],
+                                                         &los_coeffs[0])
+                    dsbis = np.repeat(Ds[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    usbis = np.repeat(us[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    ksbis = np.asarray(<double[:sz_coeff]>los_coeffs[0])
+                    # loop over time for calling and integrating
+                    for jj in range(nt):
+                        val = func(dsbis + ksbis * usbis,
+                                   t=ltime[jj], **fkwdargs)
+                        sig[jj, ii] = np.sum(val)*loc_eff_res[0]
+            elif n_imode==1:
+                for ii in range(num_los):
+                    los_coeffs = <double**>malloc(sizeof(double*))
+                    los_coeffs[0] = NULL
+                    sz_coeff = _st.LOS_get_sample_single(DLs[0,0], DLs[1,0],
+                                                         dL, n_dmode, n_imode,
+                                                         &loc_eff_res[0],
+                                                         &los_coeffs[0])
+                    dsbis = np.repeat(Ds[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    usbis = np.repeat(us[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    ksbis = np.asarray(<double[:sz_coeff]>los_coeffs[0])
+                    # loop over time for calling and integrating
+                    for jj in range(nt):
+                        val = func(dsbis + ksbis * usbis,
+                                   t=ltime[jj], **fkwdargs)
+                        sig[jj, ii] = scpintg.simps(val, x=None,
+                                                    dx=loc_eff_res[0])
+            elif n_imode==2:
+                for ii in range(num_los):
+                    los_coeffs = <double**>malloc(sizeof(double*))
+                    los_coeffs[0] = NULL
+                    sz_coeff = _st.LOS_get_sample_single(DLs[0,0], DLs[1,0],
+                                                         dL, n_dmode, n_imode,
+                                                         &loc_eff_res[0],
+                                                         &los_coeffs[0])
+                    dsbis = np.repeat(Ds[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    usbis = np.repeat(us[:,ii].reshape((3,1)), sz_coeff, axis=1)
+                    ksbis = np.asarray(<double[:sz_coeff]>los_coeffs[0])
+                    # loop over time for calling and integrating
+                    for jj in range(nt):
+                        val = func(dsbis + ksbis * usbis,
+                                   t=ltime[jj], **fkwdargs)
+                        sig[jj, ii] = scpintg.romb(val, show=False,
+                                                   dx=loc_eff_res[0])
+    # --------------------------------------------------------------------------
```

Comment:
I know all these different cases seem heavy, but I believe they are necessary to avoid a `if` statement inside a loop-loop.